packages/playwright-msw/src/utils.ts

```diff
@@ -37,5 +37,12 @@ export const getHandlerPath = (
   return (<RestHandler>handler).info.path;
 };
 
-export const convertMswPathToPlaywrightUrl = (path: Path): string | RegExp =>
-  path instanceof RegExp ? path : path.replace(/\/:[^/]+/g, '/*');
+export const convertMswPathToPlaywrightUrl = (path: Path): string | RegExp => {
+  if (path instanceof RegExp) {
+    return path;
+  }
+  const transformedPath = path.replace(/\/:[^/]+/g, '/*').replace(/\?.+/, '');
+  return transformedPath.endsWith('*')
+    ? transformedPath
+    : `${transformedPath}*`;
```

Comment:
@valendres 

It doesn't work well in case you have a trailing slash. For example, if you have `/user/:id/` it will convert it to `/user/*/*`. So in this case request to `/user/1/` won't be handled.

Why did you decide to append `*` at the end if  `transformedPath` doesn't end with `*`? If we remove it then it should work well but I didn't thought well about all possible cases.

Reply:
Ah dam, I thought it would. It seems I missed this scenario in my test cases. I added * on the end so that the glob pattern would work when query parameters are present. Unfortunately, there doesn't seem to be a way to convert `/user/:id` into a glob that works with the following urls:
1. `/user/:id`
2. `/user/:id?foo=bar`
3. `/user/:id/`
4. `/user/:id/?foo=bar`

I think the mistake here was to try and convert MSW paths to a glob. It would be much easier to convert it to a regex, which playwright supports. I'm going to work on another fix now 😅

Reply:
I've created a new PR for this: https://github.com/valendres/playwright-msw/blob/4d390af5abab4c08bb4ecb68df11e866b0f27c41/packages/playwright-msw/src/utils.test.ts#L84-L131